ui/mirage/config.js

```diff
@@ -1,581 +1,317 @@
+import ENV from 'vault/config/environment';
+import handlers from './handlers';
+
 const EXPIRY_DATE = '2021-05-12T23:20:50.52Z';
 
 export default function () {
   this.namespace = 'v1';
 
-  this.get('sys/internal/counters/activity', function (db) {
-    let data = {};
-    const firstRecord = db['clients/activities'].first();
-    if (firstRecord) {
-      data = firstRecord;
-    }
-    return {
-      data,
-      request_id: '0001',
-    };
-  });
-
-  this.get('sys/internal/counters/config', function (db) {
-    return {
-      request_id: '00001',
-      data: db['clients/configs'].first(),
-    };
-  });
+  // start ember in development running mirage -> yarn start:mirage handlerName
+  // if handler is not provided, general config will be used
+  // this is useful for feature development when a specific and limited config is required
+  const { handler } = ENV['ember-cli-mirage'];
+  if (handler && handlers[handler]) {
+    handlers[handler](this);
+    this.logging = false; // disables passthrough logging which spams the console
+    console.log(`⚙ Using ${handler} Mirage handler ⚙`); // eslint-disable-line
+  } else {
+    console.log('⚙ Using default Mirage config ⚙'); // eslint-disable-line
+    this.get('sys/internal/counters/activity', function (db) {
+      let data = {};
+      const firstRecord = db['clients/activities'].first();
+      if (firstRecord) {
+        data = firstRecord;
+      }
+      return {
+        data,
+        request_id: '0001',
+      };
+    });
 
-  this.get('/sys/internal/ui/feature-flags', function (db) {
-    const featuresResponse = db.features.first();
-    return {
-      data: {
-        feature_flags: featuresResponse ? featuresResponse.feature_flags : null,
-      },
-    };
-  });
+    this.get('sys/internal/counters/config', function (db) {
+      return {
+        request_id: '00001',
+        data: db['clients/configs'].first(),
+      };
+    });
 
-  this.get('/sys/internal/counters/activity', function () {
-    return {
-      data: {
-        start_time: '2019-11-01T00:00:00Z',
-        end_time: '2020-10-31T23:59:59Z',
-        total: {
-          distinct_entities: 200,
-          non_entity_tokens: 100,
-          clients: 300,
+    this.get('/sys/internal/ui/feature-flags', (db) => {
+      const featuresResponse = db.features.first();
+      return {
+        data: {
+          feature_flags: featuresResponse ? featuresResponse.feature_flags : null,
         },
-        by_namespace: [
-          {
-            _comment: 'by_namespace will remain as it is',
-          },
-        ],
-        months: [
-          {
-            'jan/2022': {
+      };
+    });
+
+    this.get('/sys/internal/counters/activity/monthly', function () {
```

Comment:
This is technically the response from the `/activity` endpoint - I created a `activity/monthly` response to get data to charts for that specific tab in the UI. The tab is no longer relevant in 1.10, but can be left as is for now.

Just a note @Monkeychip - we will need to update these when wiring up data with our client count work